src/runtime/pkg/magent/containerd.go

```diff
@@ -0,0 +1,94 @@
+package magent
+
+import (
+	"context"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/containers"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/typeurl"
+	"github.com/sirupsen/logrus"
+
+	vc "github.com/kata-containers/kata-containers/src/runtime/virtcontainers"
+	"github.com/kata-containers/kata-containers/src/runtime/virtcontainers/pkg/oci"
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func getContainer(containersClient containers.Store, namespace, cid string) (containers.Container, error) {
+	ctx := context.Background()
+	ctx = namespaces.WithNamespace(ctx, namespace)
+	return containersClient.Get(ctx, cid)
+}
+
+// isSandboxContainer return true if the contaienr is a snadbox container.
+func isSandboxContainer(c *containers.Container) bool {
+	// unmarshal from any to spec.
+	v, err := typeurl.UnmarshalAny(c.Spec)
+	if err != nil {
+		logrus.WithError(err).Errorf("failed to Unmarshal container spec")
+		return false
+	}
+
+	// convert to oci spec type
+	ociSpec := v.(*specs.Spec)
+
+	// get container type
+	containerType, err := oci.ContainerType(*ociSpec)
+	if err != nil {
+		logrus.WithError(err).Errorf("failed to get contaienr type")
+		return false
+	}
+
+	// return if is a sandbox container
+	return containerType == vc.PodSandbox
+}
+
+// getSandboxes get kata sandbox from containerd.
+// this should not be called too frequently
```

Comment:
It shouldn't because we have the cache in place? In reading this, I ask myself "what is too frequent?" What are your expectations? In what events would we end up calling this.

Reply:
The comments may be misleading, I have updated.
This function only run once at the startup of `magent`